### ISSUE TITLE
Is PolicyRecordingBot recieve audio from guest participant?

### ISSUE THREAD

**Issue by BohdanM19** [microsoftgraph/microsoft-graph-comms-samples#448]
@1fabi0 @ssulzer @zihzhan 
There is weird things happened if one of participant is guest. For example:
  
User **A** and **B** in organization where bot deployed. 
User **C** not in organization (connect to call as guest). 

Call 1.
User **A** and **B** - expected behaviour

Call 2.
User **A** and **C** - no audio, worker role reboot in unexpected moment, and bot clear all logs

**Logs**
All what i have is this exception
`Exception processing HTTP request.
exception: Error while copying content to a stream.

The I/O operation has been aborted because of either a thread exit or an application request
StackTrace:    at System.Net.HttpResponseStream.EndWrite(IAsyncResult asyncResult)
   at System.IO.Stream.<>c.<BeginEndWriteAsync>b__53_1(Stream stream, IAsyncResult asyncResult)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncTrimPromise`1.Complete(TInstance thisRef, Func`3 endMethod, IAsyncResult asyncResult, Boolean requiresSynchronization)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Owin.Host.HttpListener.RequestProcessing.ExceptionFilterStream.<WriteAsync>d__36.MoveNext()`

Is someone know, there is oportunity to record guest user audio? Is this is application error or platform restrictions

**Comment by 1fabi0** [comment]
Checkout this Stack Overflow answer https://stackoverflow.com/a/35730278/12799021 and make sure your not reading somwhere twice from the streams in the Controller action for Roaster Updates

**Comment by dannygar** [comment]
Yes, the Compliance Policy Bot can receive audio from both tenant and outside-org users. The policy cannot be assigned to anyone outside of the tenant though.
The exception you're seeing most likely happen in the application level, and my guess it is when the Bot is trying to retrieve the participant info from the meeting participants list. There is a difference in how to fetch the participant info when it comes to a tenant user vs a guest. 
For example, if you need to obtain the `Identity` object from the `participant` object, you should check for null in this line:
```c#
var user = participant.Resource.Info.Identity.User ?? participant.Resource.Info.Identity.GetGuest();
```
as you can see, the Identity for the tenant user is fetched from the `User` property, when the identity of the guest is fetched from the `GetGuest()` method. As soon as you get the right `user` object, you can retrieve the participant's Id, Display Name and other information
